XiEditor/EditViewController.swift

```diff
@@ -50,6 +50,8 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
 
     @IBOutlet weak var editViewHeight: NSLayoutConstraint!
     @IBOutlet weak var editViewWidth: NSLayoutConstraint!
+    
+    let appDelegate = NSApp.delegate as! AppDelegate
```

Comment:
Not sure if this is the right way to access AppDelegate. Not even sure if I'm supposed to access the AppDelegate in a VC.

Reply:
I would avoid storing the variable, and instead just call `(NSApp.delegate as? AppDelegate)?.whatever()` in line, as needed.

Reply:
Thanks for your feedback on this, always interesting to see various practices!

Reply:
It's generally not good practice to reference the AppDelegate inside of your view controllers (directly).

How about using a protocol?

e.g.

```swift
protocol ThemeChangeDelegate {
    func themeChanged(name: String, theme: Theme)
}


...

extension AppDelegate: ThemeChangeDelegate { }

...

weak var themeChangeDelegate: ThemeChangeDelegate? = NSApp.delegate as? ThemeChangeDelegate

```

It will make it much easier to write a unit test and decouples the editor view.

Reply:
Adding to what @ollieatkinson wrote, the best option is to use dependency injection to add the delegate into the view controller:

Here, the main view controller is setting its `themeChangeDelegate` on the `ThemeViewController`, so it is just copying its dependency into the `ThemeViewController`
``` swift
class MainViewController: NSViewController {
  private var themeViewController: ThemeViewController?
  private var themeChangeDelegate: ThemeChangeDelegate?
  override func viewDidLoad() {
    super.viewDidLoad()
    self.themeViewController = ThemeViewController()
    self.themeViewController?.themeChangeDelegate = self.themeChangeDelegate
  }
}
```

Here, the AppDelegate sets up the `MainViewController` and installs itself as the dependency. That way, nobody ever accesses the AppDelegate except for the appdelegate itself.
``` swift
class AppDelegate: NSObject, NSApplicationDelegate, ThemeChangeDelegate {
  var mainViewController = MainViewController()
  func applicationWillFinishLaunching(_ aNotification: Notification) {
    mainViewController.themeChangeDelegate = self
  }
}
```